src/components/nodes/boundaryEvent/boundaryEvent.vue

```diff
@@ -10,10 +10,43 @@ import { id as taskId } from '@/components/nodes/task';
 import { id as callActivityId } from '@/components/nodes/callActivity';
 import { id as manualTaskId } from '@/components/nodes/manualTask';
 import { id as scriptTaskId } from '@/components/nodes/scriptTask';
+import portsConfig from '@/mixins/portsConfig';
+import { portGroups } from '@/mixins/portsConfig';
+import { g } from 'jointjs';
+
+function getPointFromGroup(model, group) {
+  const { x: shapeX, y: shapeY } = model.position();
+  const { x, y } = Object.values(model.getPortsPositions(group))[0];
+  return new g.Point(shapeX + x, shapeY + y);
+}
+
+function getPortPoints(model) {
+  return portGroups.filter(group => group !== 'absolute').map(group => getPointFromGroup(model, group));
+}
+
+function closestPort(model, anchorReference) {
+  return getPortPoints(model).sort((p1, p2) => {
+    const referencePoint = new g.Point(anchorReference.x, anchorReference.y);
+    return referencePoint.distance(p1) - referencePoint.distance(p2);
+  })[0];
+}
+
+function hasPorts(model) {
+  return Object.values(model.getPortsPositions(portGroups[0])).length > 0;
+}
+
+function snapToAnchor(coords, model) {
+  if (!hasPorts(model)) {
+    const { x, y } = model.position();
+    const { width, height } = model.size();
+    return new g.Point(x - (width / 2), y - (height / 2));
+  }
+  return closestPort(model, coords);
+}
```

Comment:
I really like all of the code in these functions: it seems well factored, named and expressive. I'm wondering why it written as functions inside this module file though. Is there a reason why you didn't want to make them methods of the component? If it's because you think they collectively represent a unit of functionality (which they seem to) how about moving them into another file? You could them just export `snapToAnchor()`, which seems to be the single entry point into this functionality.

Reply:
What about making it a mixin?